51.md

```diff
@@ -6,148 +6,153 @@ Lists
 
 `draft` `optional` `author:fiatjaf` `author:arcbtc` `author:monlovesmango` `author:eskema` `author:gzuuus`
 
-A "list" event is defined as having a list of public and/or private tags. Public tags will be listed in the event `tags`. Private tags will be encrypted in the event `content`. Encryption for private tags will use [NIP-04 - Encrypted Direct Message](04.md) encryption, using the list author's private and public key for the shared secret. A distinct event kind should be used for each list type created.
+This NIP defines lists of things that users can create. Lists can contain references to anything, and these
+references can be **public** or **private**.
 
-If a list should only be defined once per user (like the "mute" list) the list is declared as a _replaceable event_. These lists may be referred to as "replaceable lists". Otherwise, the list is a _parameterized replaceable event_ and the list name will be used as the `d` tag. These lists may be referred to as "parameterized replaceable lists".
+Public items in a list are specified in the event `tags` array, while private items are specified in a JSON
+array that mimics the structure of the event `tags` array, but stringified and encrypted using the same
+scheme from [NIP-04](04.md) (the shared key is computed using the author's public and private key) and
+stored in the `.content`.
 
-## Replaceable List Event Example
+## Types of lists
 
-Lets say a user wants to create a 'Mute' list and has keys:
-```
-priv: fb505c65d4df950f5d28c9e4d285ee12ffaf315deef1fc24e3c7cd1e7e35f2b1
-pub: b1a5c93edcc8d586566fde53a20bdb50049a97b15483cb763854e57016e0fa3d
-```
-The user wants to publicly include these users:
+### Generic lists
 
-```json
-["p", "3bf0c63fcb93463407af97a5e5ee64fa883d107ef9e558472c4eb9aaaefa459d"],
-["p", "32e1827635450ebb3c5a7d12c1f8e7b2b514439ac10a67eef3d9fd9c5c68e245"]
-```
-and privately include these users (below is the JSON that would be encrypted and placed in the event content):
+The kind `30001` has been reserved for generic lists. These must be accompanied by a `d` tag identifying the
+list, but these have no standard meaning and are generally client-specific (except in the standard cases
+specified below).
 
-```json
-[
-    ["p", "9ec7a778167afb1d30c4833de9322da0c08ba71a69e1911d5578d3144bb56437"],
-    ["p", "8c0da4862130283ff9e67d889df264177a508974e2feb96de139804ea66d6168"]
-]
-```
+## Standard lists
 
-Then the user would create a 'Mute' list event like below:
+Users are expected to have a single list of each of these types. They have special meaning and clients may
+rely on them to augment the user profile or browsing experience.
```

Comment:
`d` should not be both client-defined *and* user-defined on the same kind, because that results in an overloaded namespace. What if a user creates a list called "communities"? I suppose `d` could be a uuid or something instead. But even then, clients that only want to show user-defined lists have to whitelist the standard `d` tags. Standard lists should be different kinds: 10001 for bookmarks, 10002 for pins, 10003 for communities. This is how the old spec was written, although those kinds weren't specified and people ended up using 30001 for them:

> ### Categorized Bookmarks List
> 
> An event of kind `30001` is defined as a parameterized replaceable list event for categorizing bookmarks. The 'd' parameter for this event holds the category name of the list.

"The 'd' parameter for this event holds the category name of the list." This is how Coracle uses kind 30001 — by setting the `d` tag to the `title` of the list (I didn't actually know about title or description, I don't know when they were introduced). 

Can we not break my correct implementation because of widespread incorrect ones?